src/python/grpcio/grpc/_server.py

```diff
@@ -670,6 +670,10 @@ def _on_call_completed(state):
 
 def _serve(state):
     while True:
+        # We want to force the deletion of the last event         
+        # ~before~ we poll again, if the event has a reference  
+        # to a shutdown Call object, this can induce spinlock
+        event = None
```

Comment:
I think this would be more idiomatic at the end of the loop rather than at the beginning, so that the reader can see that it is useful every time it is executed (whereas when it appears at the beginning of the loop it is silly and meaningless the first time it is executed and helpful every other time).